protocol/lavaprotocol/finalization_consensus.go

```diff
@@ -276,3 +308,123 @@ func InterpolateBlocks(timeNow, latestBlockTime time.Time, averageBlockTime time
 	}
 	return int64(timeNow.Sub(latestBlockTime) / averageBlockTime) // interpolation
 }
+
+func VerifyFinalizationData(reply *pairingtypes.RelayReply, relayRequest *pairingtypes.RelayRequest, providerAddr string, consumerAcc sdk.AccAddress, latestSessionBlock, blockDistanceForFinalization, blocksInFinalizationProof int64) (finalizedBlocks map[int64]string, errRet error) {
+	relayFinalization := conflicttypes.NewRelayFinalizationFromRelaySessionAndRelayReply(relayRequest.RelaySession, reply, consumerAcc)
+	recoveredProviderPubKey, err := sigs.RecoverPubKey(relayFinalization)
+	if err != nil {
+		return nil, err
+	}
+
+	recoveredProviderAddr, err := sdk.AccAddressFromHexUnsafe(recoveredProviderPubKey.Address().String())
+	if err != nil {
+		return nil, err
+	}
+
+	if recoveredProviderAddr.String() != providerAddr {
+		return nil, utils.LavaFormatError("provider address mismatch in finalization data ", ProviderFinalizationDataError,
```

Comment:
this should be some sort of a conflict as well. can you create a ticket (with enough explanation) so we dont forget to deal with this case?

Reply:
https://lavanet.atlassian.net/browse/PRT-1185?atlOrigin=eyJpIjoiMWFjNjUyOGZhYjQwNGFiYWE4ZmE2MmU5ZTAzYjdjYzAiLCJwIjoiaiJ9